libr/anal/anal.c

```diff
@@ -690,3 +692,41 @@ R_API bool r_anal_is_prelude(RAnal *anal, const ut8 *data, int len) {
 	}
 	return false;
 }
+
+R_API void r_anal_add_import(RAnal *anal, const char *imp) {
+	if (anal->imports) {
+		RListIter *it;
+		const char *eimp;
+		r_list_foreach (anal->imports, it, eimp) {
+			if (!strcmp (eimp, imp)) {
+				return;
+			}
+		}
+	} else {
+		anal->imports = r_list_newf (free);
```

Comment:
Why did you opt for lazy allocation of the list? Can't we just allocate `anal->imports` when creating `RAnal` object like it's done for other stuff and in the `r_anal_purge_imports` just do a `r_list_purge`.

Reply:
Because it was like that before, this pr does not change the approach (yet).

Reply:
yeah I saw that later. I think it makes sense to do this change directly here if you want.

Reply:
Agreed.